src/components/PostDetails.tsx

```diff
@@ -1,106 +1,121 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Comment, CommentData } from '../types/Comment';
+import { Post } from '../types/Post';
+import { User } from '../types/User';
+
+type Props = {
+  showCommentLoader: boolean;
+  postComments: Comment[];
+  selectedPost: Post | null;
+  errorsComment: boolean;
+  handleDeleteComment: (commentId: number) => void;
+  addComment: (newComment: CommentData) => void;
+  buttonAddLoading: boolean;
+  setShowCommentForm: (item: boolean) => void;
```

Comment:
The method `setShowCommentForm` should be named `onShowCommentFormChange` to follow the recommended naming conventions.